crates/hir-ty/src/chalk_db.rs

```diff
@@ -884,13 +927,378 @@ fn impl_def_datum(db: &dyn HirDatabase, krate: Crate, impl_id: hir_def::ImplId)
     Arc::new(impl_datum)
 }
 
+fn impl_rpitit_values(
```

Comment:
I feel this shouldn't live here in `chalk_db` but rather in its own module, since it's not just some Chalk plumbing